collect/collect.go

```diff
@@ -291,6 +293,15 @@ func (i *InMemCollector) checkAlloc(ctx context.Context) {
 	tracesSent := generics.NewSet[string]()
 	// Send the traces we can't keep.
 	for _, trace := range allTraces {
+		if !i.IsMyTrace(trace.ID()) {
+			err := errors.New("cannot make a decision for partial traces")
+
+			i.Logger.Warn().WithFields(map[string]interface{}{
```

Comment:
Isn't this really common? Do we actually want to be sending thousands of copies of this warning? 

If we keep it, I think the error message could be clearer (what kind of decision are we trying to make?).

And finally, it's a constant string -- why construct an error out of it, and then convert it back to a string?

Reply:
I agree with you concern. I changed the log level to debug and the message to be more specific